### PR TITLE
task: add a releasing script

### PR DESCRIPTION
**Pull request by gssbzn**
## Proposed changes

Simplify the release process 

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -10,8 +10,15 @@ these tasks are run on master and can be patched at any time.
...
+./scripts/release.sh 1.0.0
+```
+
+**Note:** Please omit the `v` from the version to release 
+
+This will do the following things:
 1. Tag a new version, ie: `git tag -a -s v1.0.0 -m "v1.0.0"`
 2. Publish the new tag, ie `git push origin v1.0.0`
-3. Run the evergreen releasing task, ie: `evergreen patch -p mongocli-master -y -d "Release v1.0.0" -v release_publish -v release_msi -t all`
+3. Run the evergreen releasing task, ie: `evergreen patch -p mongocli-master -y -d "Release v1.0.0" -v release_publish -v release_msi -t all -f`
```
Comment (colm-quinn) [inline RELEASING.md:24]: 👍

File: RELEASING.md [hunk 2]
```diff
@@ -10,8 +10,15 @@ these tasks are run on master and can be patched at any time.
...
 ## Stable release
 
-To generate a new stable release you must follow the following steps:
+To generate a new stable release you can run:
 
+```bash
+./scripts/release.sh 1.0.0
+```
+
+**Note:** Please omit the `v` from the version to release
```
Comment (colm-quinn) [inline RELEASING.md:19]: [np] Could we check and do this in the script? It's counter-intuitive as the examples below all show the v present.

File: scripts/release.sh [hunk 3]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
```
Comment (colm-quinn) [inline scripts/release.sh:3]: [q] Would `-x` be useful as well for debugging or we don't care?

Comment (gssbzn) [reply scripts/release.sh:3]: if I add it you'll see the script running
```bash
./scripts/release.sh 123
+ VERSION=123
+ [[ -z 123 ]]
+ read -p 'Are you sure to release v123? ' -n 1 -r
Are you sure to release v123?
```
a bit ugly IMO

File: RELEASING.md (hunk 2 above)
Comment (gssbzn) [reply RELEASING.md:19]: Thought about it but was being lazy, I'll do the full work

File: scripts/release.sh (hunk 3 above)
Comment (colm-quinn) [reply scripts/release.sh:3]: Yeah that's fair!

File: scripts/release.sh [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+if [[ -z "${VERSION}" ]]; then
+    echo "Please provide a version"
+    exit 1
+fi
+
+if [[ "${VERSION}" == v* ]]; then
+    echo "Please omit the 'v' when using this script"
+    exit 1
+fi
```
Comment (colm-quinn) [inline scripts/release.sh:15]: 👍

**Pull request merged**